### PR TITLE
A bunch of cleanups and a missing `ShortcutRegistar` in `WidgetsApp`

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

A bunch of random cleanup things I found while doing `MenuBar` development.
- Changes an if test to an assert in `binding.dart`, since the if should always be true.
- Adds the default `ShortcutRegistrar` that should have been in the `ShortcutRegistry` PR.
- Moves a debug message in the `FocusManager` to print the result after the focus change instead of before.
- Reorders the test parameters in theme_data_test.dart to match the order of the theme data fields everywhere else.

## Tests
- Added a test for the missing `ShortcutRegistar` in `WidgetsApp`.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/app.dart [hunk 1]
```diff
@@ -1730,7 +1730,9 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
               actions: widget.actions ?? WidgetsApp.defaultActions,
               child: FocusTraversalGroup(
                 policy: ReadingOrderTraversalPolicy(),
-                child: child,
+                child: ShortcutRegistrar(
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/widgets/app.dart:1733]: This change does not look like a clean up. Is it intended?

Comment (dkwingsmt) [reply packages/flutter/lib/src/widgets/app.dart:1733]: Ok I see you mentioning it in the PR. Maybe this should be the title of this PR.

Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/app.dart:1733]: You're right, it's not really a "cleanup", but something that was missing from the previous PR that I committed with the `ShortcutRegistry`.  Without this, people will have to add their own registrar manually.

I've added a test to make sure it exists.

**Review by dkwingsmt — APPROVED** [review]
LGTM

**Review by cbracken — APPROVED** [review]

**Pull request merged**